rs/consensus/src/consensus/purger.rs

```diff
@@ -455,16 +457,47 @@ fn get_purge_height(pool_reader: &PoolReader<'_>) -> Option<Height> {
         })
 }
 
+/// Return the heights of all finalized IDKG summary blocks for which
+/// there exists no CUP artifact yet.
+fn get_pending_idkg_cup_heights(pool: &PoolReader<'_>) -> BTreeSet<Height> {
+    let mut pending_cup_heights = BTreeSet::new();
+    let current_cup_height = pool.get_catch_up_height();
+    let mut block = pool.get_highest_summary_block();
+    while block.height() > current_cup_height {
```

Comment:
Practically, `highest_summary_block` is the only one we really need to check if there is a CUP or not because Consensus stalls if the gap between finalized block and the summary block without CUP reaches limit. 

Therefore, I prefer to give `get_highest_summary_block` a special treatment in the code as it always return the `Block` we care without any errors or `None` case unlike `get_finalized_block`.  WDYT? @eichhorl

Reply:
Reversing the order does seem cleaner, and it saves one call to `get_finalized_block` in the practical case

Reply:
If there are no additional changes to address, would it be alright if I go ahead and merge the PR?

Reply:
LGTM, sorry for the delay

Reply:
Thank you for the review!